File: src/extractors/sites/tiktok_extractor.cpp
#include "tiktok_extractor.hpp"
#include "../../config.hpp"

bool TikTokExtractor::matches(
    const std::string& url
)
{
    return
        url.find("tiktok.com")
        != std::string::npos;
}

std::vector<DownloadTask>
TikTokExtractor::extract(
    const std::string& url
)
{
    DownloadTask task;

    task.url = url;

    task.use_ytdlp = true;

    task.format_id =
        Config::get_default_format();

    return { task };
}

File: android/app/src/main/java/com/mediadownloader/extractors/sites/TikTokExtractor.kt
package com.mediadownloader.extractors.sites

import com.mediadownloader.config.Config
import com.mediadownloader.extractors.BaseExtractor
import com.mediadownloader.extractors.DownloadTask

class TikTokExtractor : BaseExtractor {
    override fun matches(url: String): Boolean =
        url.contains("tiktok.com")

    override suspend fun extract(url: String): List<DownloadTask> =
        listOf(
            DownloadTask(
                url = url,
                useYtdlp = true,
                formatId = Config.getDefaultFormat()
            )
        )
}
